compiler/compile/OMRSymbolReferenceTable.hpp

```diff
@@ -186,11 +186,65 @@ class SymbolReferenceTable
        */
       synchronizedFieldLoadSymbol,
 
-      // common atomic primitives
+      /** \brief
+       *
+       *  This symbol represents an intrinsic call of the following format:
+       *
+       *  \code
+       *    icall <atomicAdd32BitSymbol>
+       *      <address>
+       *      <value>
+       *  \endcode
+       *
+       *  Which performs the following operation atomically:
+       *
+       *  \code
+       *    [address] = [address] + <value>
+       *    return <value>
+       *  \endcode
+       */
       atomicAdd32BitSymbol,
       atomicAdd64BitSymbol,
```

Comment:
@0xdaryl @0dvictor personally I don't think we should distinguish between 32-bit and 64-bit since this can be implied by the data type of `<value>`. I'm in favor of folding these into one `atomicAddSymbol` and doing a type check on the second child at codegen time to determine whether the addition is 32-bit or 64-bit. Is there any downside to this? I don't want to bloat these symbol tables with unnecessary enums.

Reply:
Fixed in 5da969d and 8d30aae